server/src/domain/person/person.service.ts

```diff
@@ -241,62 +243,54 @@ export class PersonService {
     return results;
   }
 
-  async handlePersonDelete({ id }: IEntityJob) {
-    const person = await this.repository.getById(id);
-    if (!person) {
-      return false;
-    }
-
-    try {
-      await this.repository.delete(person);
-      await this.storageRepository.unlink(person.thumbnailPath);
-    } catch (error: Error | any) {
-      this.logger.error(`Unable to delete person: ${error}`, error?.stack);
-    }
-
-    return true;
+  private async delete(people: PersonEntity[]) {
+    await Promise.all(people.map(person => this.storageRepository.unlink(person.thumbnailPath)));
+    await this.repository.delete(people);
+    this.logger.debug(`Deleted ${people.length} people`);
   }
 
   async handlePersonCleanup() {
     const people = await this.repository.getAllWithoutFaces();
-    for (const person of people) {
-      this.logger.debug(`Person ${person.name || person.id} no longer has any faces, deleting.`);
-      await this.jobRepository.queue({ name: JobName.PERSON_DELETE, data: { id: person.id } });
-    }
-
+    await this.delete(people);
     return true;
   }
 
-  async handleQueueRecognizeFaces({ force }: IBaseJob) {
+  async handleQueueDetectFaces({ force }: IBaseJob) {
     const { machineLearning } = await this.configCore.getConfig();
     if (!machineLearning.enabled || !machineLearning.facialRecognition.enabled) {
       return true;
     }
 
+    if (force) {
+      const people = await this.repository.getAll();
+      await this.delete(people); // deletes thumbnails too
+      await this.repository.deleteAllFaces();
+    }
```

Comment:
This should probably be done in batches

Reply:
This was more work than I thought, but it's done